core_helpers/logging_objects.py

```diff
@@ -83,3 +95,49 @@ def failure(self,msg,level='info',pre_format='FAIL: '):
     def set_rp_logger(self,rp_pytest_service):
         "Set the reportportal logger"
         self.rp_logger = self.log_obj.setup_rp_logging(rp_pytest_service)
+
+    def conditional_write(self,flag,positive,negative,level='info'):
+        "Write out either the positive or the negative message based on flag"
+        self.mini_check_counter += 1
+        if level.lower() == "inverse":
+            if flag is True:
+                self.write(positive,level='error')
+            else:
+                self.write(negative,level='info')
+                self.mini_check_pass_counter += 1
+        else:
+            if flag is True:
+                self.write(positive,level='info')
+                self.mini_check_pass_counter += 1
+            else:
+                self.write(negative,level='error')
+
+    def make_gif(self):
+        "Create a gif of all the screenshots within the screenshots directory"
+        self.gif_file_name = Gif_Maker.make_gif(self.screenshot_dir,name=self.calling_module)
+
+        return self.gif_file_name
+
+    def set_calling_module(self,name):
+        "Set the test name"
+        self.calling_module = name
+
+    def get_calling_module(self):
+        "Get the name of the calling module"
+        if self.calling_module is None:
+            #Try to intelligently figure out name of test when not using pytest
+            full_stack = inspect.stack()
```

Comment:
Do we need these steps to get the test name? We now have a `testname` fixture in `conftest.py`, we use it to name our log files. Can we propagate the name to the `Logging_Objects` class?

Reply:
The test_name is already propagated to logging_objects via the `set_calling_module` method. The `get_calling_module(self)`, executes those additional steps to get the testname only if the `calling_module` is None.
Do we need to remove the `get_calling_module` method itself?

Reply:
Did you check in what cases the `set_calling_module` will return `None`? the `testname` fixture in `conftest.py` file will always return the `testname`

Reply:
You are right, the conftest  fixture will never return none. I have modified the `get_calling_module` method